CMakeLists.txt

```diff
@@ -151,7 +151,40 @@ if(BML_MPI)
   add_definitions(-DDO_MPI)
 endif()
 
-set(BML_CUDA FALSE CACHE BOOL "Whether to compile with CUDA support")
+option(BML_GPU "Compile with OpenMP GPU Offload support" FALSE)
+if(BML_GPU)
+  message(STATUS "Will build with OpenMP GPU Offload support")
+  add_definitions(-DDO_GPU)
+
+  find_package(OpenMP 4.5 REQUIRED)
+  set(OpenMP_CXX_VERSION ${OpenMP_CXX_VERSION} CACHE STRING "OpenMP CXX Version")
+
+  include(CheckCXXCompilerFlag)
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(OpenMP_CXX_OFFLOAD_FLAG "-foffload='-lm'")
+  elseif(MAKE_CXX_COMPILER_ID STREQUAL "Intel")
+    set(OpenMP_CXX_OFFLOAD_FLAG "-foffload='-lm'")
+  elseif(MAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    set(OpenMP_CXX_OFFLOAD_FLAG "-fopenmp-targets=nvptx64-nvidia-cuda --cuda-path=${CUDA_TOOLKIT_ROOT_DIR}")
+  endif()
+  check_cxx_compiler_flag("${OpenMP_CXX_OFFLOAD_FLAG}" OFFLOAD_CXX_FOUND)
+  if(OFFLOAD_CXX_FOUND)
+    message("Found offload")
+  endif()
+  set(OpenMP_CXX_FLAGS "${OpenMP_CXX_FLAGS} ${OpenMP_CXX_OFFLOAD_FLAG}")
+  set(OpenMP_C_FLAGS "${OpenMP_C_FLAGS} ${OpenMP_CXX_OFFLOAD_FLAG}")
+  set(OpenMP_Fortran_FLAGS "${OpenMP_Fortran_FLAGS} ${OpenMP_CXX_OFFLOAD_FLAG}")
+
+  list(APPEND LANGUAGES CUDA)
+  set(CUDA_TOOLKIT_ROOT_DIR ${CUDA_TOOLKIT_ROOT_DIR})
+  include(FindCUDA)
+  set(GPU_ARCH "sm_60" CACHE STRING "LAMMPS GPU CUDA SM architecture")
+  set_property(CACHE GPU_ARCH PROPERTY STRINGS sm_30 sm_50 sm_60 sm_70)
+
+  message("CUDA libraries: ${CUDA_CUBLAS_LIBRARIES} xxx ${CUDA_LIBRARIES}")
```

Comment:
I guess the part behind `xxx` is for debug!

Reply:
I'll remove it